subsys/net/ip/net_shell.c

```diff
@@ -2698,12 +2718,14 @@ static enum net_verdict handle_ipv6_echo_reply(struct net_pkt *pkt,
 
 static int ping_ipv6(const struct shell *shell,
 			char *host,
+			unsigned int size,
 			unsigned int count,
 			unsigned int interval)
 {
 	struct in6_addr ipv6_target;
 	struct net_if *iface = net_if_get_default();
 	struct net_nbr *nbr;
+	uint8_t payload[size += sizeof(u32_t)];
```

Comment:
Hi @benpicco ,
About this payload, is it allocated from heap or thread stack or something else?
If from thread stack, is it possible to be overflow with big size?

Reply:
Using VLAs is no good, please do it differently. One option could be to allocate enough net_buf's for a given size, and then pass that chain of net_buf's directly to `net_icmpv6_send_echo_request()` which can then send that data directly.

Reply:
Yes it's allocated on the stack and yes that's bad.

I'm not sure how to do it with net_bufs though. Is there a way where I don't have to reserve dedicated memory for it using `NET_BUF_POOL_DEFINE` but instead can make use of some existing pool?

Another way would be to only enable the -s option if `CONFIG_HEAP_MEM_POOL_SIZE` is not 0.

Reply:
Hmm, we seem to have lost proper API in net_pkt.h for allocating net_buf's.

Another option could be to use `net_pkt_skip()` and optionally with `net_pkt_memset()`. That way we could fill enough net_buf's with given byte value. This would be needed to be done inside `net_icmpv6_send_echo_request()` as the net_pkt is not exposed outside of that function.

So for example we could have something like `net_icmpv6_send_echo_request(iface, dst, id, seq, len, value)`, and inside that func just do `net_pkt_skip(pkt, len)` followed by cursor setting and then `net_pkt_memset(pkt, value)`. The cursor setting is a bit vague here as I am just brainstorming atm.

Reply:
After looking how net_pkt_skip() and net_pkt_memset() are implemented, it is probably enough just to call net_pkt_memset() which should allocate enough net_buf's to fill the packet.